src/uct/ib/dc/dc_mlx5.c

```diff
@@ -968,20 +976,32 @@ ucs_status_t uct_dc_handle_failure(uct_ib_iface_t *ib_iface, uint32_t qp_num,
     uct_dc_mlx5_iface_dci_put(iface, dci);
     ucs_assert_always(ep->dci == UCT_DC_MLX5_EP_NO_DCI);
 
-    ep_status = iface->super.super.super.ops->set_ep_failed(ib_iface,
-                                                      &ep->super.super, status);
-    if (ep_status == UCS_OK) {
-        status = uct_dc_mlx5_iface_reset_dci(iface, dci);
-        if (status != UCS_OK) {
-            ucs_fatal("iface %p failed to reset dci[%d] qpn 0x%x: %s",
-                       iface, dci, txqp->qp->qp_num, ucs_status_string(status));
+    if (ep == iface->tx.fc_ep) {
+        /* Cannot handle errors on flow-control endpoint.
```

Comment:
Shall not we destroy fc_ep and set all real related EPs to failed?

Reply:
i guess you mean find eps with same address? currently we don't hold list of dc_eps and adding such list would increase dc memory footprint... also don't see point to destroy fc_ep since it's not really bound to a particular destination
we could just ignore the error and reset the dci, and if there is another ep to same peer it would fail on its own anyway

re why is needed in UCP - to make sure we get transport-level error message in the next case we wouldn't handle it properly..